### PR TITLE
allow bump.sh to work on darwin

### PR DESCRIPTION
**Pull request by spiffxp**
good ol `sed -i` on darwin, and upcase/downcase shorthand that doesn't work on the default bash version on darwin

### CODE REVIEW

**Comment by rmmh** [comment]
can you just brew install bash? :(

File: prow/bump.sh [hunk 1]
```diff
@@ -18,15 +18,28 @@ if [ "$#" -ne 1 ]; then
...
+if ! ($SED --version 2>&1 | grep -q GNU); then
+  echo "!!! GNU sed is required.  If on OS X, use 'brew install gnu-sed'."
+  exit 1
+fi
+
 cd $(dirname $0)
 
-makefile_version_re="^\(${1^^}_VERSION.*=\s*\)"
-version=$(sed -n "s/$makefile_version_re//p" Makefile)
+program_lower=$(echo "$1" | tr "A-Z" "a-z")
```
Comment (rmmh) [inline prow/bump.sh]: can we use case insensitive seds instead?

**Comment by spiffxp** [comment]
@rmmh I'm happy to switch `/bin/bash` to `/usr/bin/env bash` if that's more palatable, there's some prior art for that in kubernetes/kubernetes, but the majority of scripts in there default to `/bin/bash`

The `sed -i` workaround I don't know a better way to do

File: prow/bump.sh [hunk 2]
```diff
@@ -18,15 +18,28 @@ if [ "$#" -ne 1 ]; then
...
+program_upper=$(echo "$1" | tr "a-z" "A-Z")
+
+makefile_version_re="^\(${program_upper}_VERSION.*=\s*\)"
+version=$($SED -n "s/$makefile_version_re//p" Makefile)
```
Comment (rmmh) [inline prow/bump.sh]: this would become `//Ip`

File: prow/bump.sh [hunk 3]
```diff
@@ -18,15 +18,28 @@ if [ "$#" -ne 1 ]; then
...
 
-echo "program: $1"
+echo "program: $program_lower"
 echo "old version: $version"
 echo "new version: $new_version"
 
-sed -i "s/$makefile_version_re.*/\1$new_version/" Makefile
-sed -i "s/\(${1,,}:\)[0-9.]*/\1$new_version/" cluster/*.yaml
+$SED -i "s/$makefile_version_re.*/\1$new_version/" Makefile
+$SED -i "s/\(${program_lower}:\)[0-9.]*/\1$new_version/" cluster/*.yaml
```
Comment (rmmh) [inline prow/bump.sh]: similarly, add I to the end of these

File: prow/bump.sh (hunk 1 above)
Comment (rmmh) [inline prow/bump.sh]: that way you only need program=$1, not $program_lower and $program_upper

**Pull request merged**